src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Buttons/BitToggleButtonTests.cs

```diff
@@ -0,0 +1,195 @@
+            DataRow(Visual.Material, false, false, "Button checked", "Button unchecked", "EmojiNeutral", "Emoji2", false)
+        ]
+        public Task BitToggleButtonClickEvent(Visual visual,
+            bool isChecked, bool isEnabled,
+            string checkedLabel, string unCheckedLabel, string checkedIconName, string unCheckedIconName,
+            bool expectedResult)
+        {
+            var component = RenderComponent<BitToggleButtonTest>(parameters =>
+            {
+                parameters.Add(p => p.Visual, visual);
+                parameters.Add(p => p.IsChecked, isChecked);
+                parameters.Add(p => p.ChekedLabel, checkedLabel);
+                parameters.Add(p => p.UnChekedLabel, unCheckedLabel);
+                parameters.Add(p => p.CheckedIconName, checkedIconName);
+                parameters.Add(p => p.UnCheckedIconName, unCheckedIconName);
+                parameters.Add(p => p.IsEnabled, isEnabled);
+            });
+
+            var bitToggleButton = component.Find(".bit-tgl-btn");
+
+            bitToggleButton.Click();
+
+            Assert.AreEqual(isEnabled ? 1 : 0, component.Instance.CurrentCount);
+
+            Assert.AreEqual(expectedResult, bitToggleButton.ClassList.Contains("bit-tgl-btn-checked"));
+
+            var bitIconTag = component.Find(".bit-tgl-btn > span > i");
+
+            Assert.AreEqual(isEnabled, bitIconTag.ClassList.Contains($"bit-icon--{(isChecked ? unCheckedIconName : checkedIconName)}"));
+
+            var bitLabelTag = component.Find(".bit-tgl-btn > span > span");
+            if (isEnabled)
+            {
+                Assert.AreEqual(bitLabelTag.TextContent, (isChecked ? unCheckedLabel : checkedLabel));
+            }
+            else
+            {
+                Assert.AreEqual(bitLabelTag.TextContent, (isChecked ? checkedLabel : unCheckedLabel));
+            }
+
+            return Task.CompletedTask;
+        }
+
+        [DataTestMethod,
+          DataRow(true, ButtonStyle.Primary, false, false),
+          DataRow(true, ButtonStyle.Standard, true, false),
+          DataRow(false, ButtonStyle.Primary, false, true),
+          DataRow(false, ButtonStyle.Standard, true, false),
+      ]
+        public Task BitToggleButtonDisabledFocusTest(bool isEnabled, ButtonStyle style,
+            bool allowDisabledFocus, bool expectedResult)
+        {
+            var com = RenderComponent<BitToggleButtonTest>(parameters =>
+            {
+                parameters.Add(p => p.IsEnabled, isEnabled);
+                parameters.Add(p => p.ButtonStyle, style);
+                parameters.Add(p => p.AllowDisabledFocus, allowDisabledFocus);
+            });
+
+            var bitButton = com.Find(".bit-tgl-btn");
+
+            var hasTabindexAttr = bitButton.HasAttribute("tabindex");
+
+            Assert.AreEqual(hasTabindexAttr, expectedResult);
```

Comment:
again, instead of introducing the "expectedResult" argument, try to calculate this result by implementing the actual logic in this test method.